_posts/SieveOfEratosthenes.md

```diff
@@ -0,0 +1,132 @@
+            isPrime[ i ] = false;
+        }
+        for ( int p = 3; p <= LIMIT; p += 2 ) {
+            if ( isPrime[ p ] ) {
+                primes.add( p );
+                for ( int m = 3; m * p < LIMIT; m += 2 ) {
+                    isPrime[ m * p ] = false;
+                }
+            }
+        }
+    }
+
+
+## Pseudo-Code for Selection Sort
+private static final boolean[] isPrime = new boolean[121];
+    static {
+        Arrays.fill( isPrime, true ); // first we assume all numbers are primes if not shown otherwise
+        isPrime[ 0 ] = false; // zero is not prime - it'is not greater than 1
+        isPrime[ 1 ] = false; // one is not prime - it'is not greater than 1
+        for ( int p = 2; p < isPrime.length; ++p ) {
+            if ( isPrime[ p ] ) {
+                for ( int m = 2; m * p < 121; ++m ) {
+                    isPrime[ m * p ] = false;
+                }
+            }
+        }
+    }
+Now, you can answer the question "is n prime" in constant time. Of course problems like this one are so simple, that you won't see it as problems, but it is subproblem of some more diffucult problem often.
+
+You can memorize the primes too. And while it holds that even number multiply by whatever is even (and there is just one even prime number), you can skip all even m:
+
+## Implementation of Sieve of Eratosthenes in Java
+ private static final int LIMIT = 121;
+    private static final boolean[] isPrime = new boolean[LIMIT + 1];
+    private static ArrayList<Integer> primes = new ArrayList<Integer>();
+    static {
+        Arrays.fill( isPrime, true ); // first we assume all numbers are primes if not shown otherwise
+        isPrime[ 0 ] = false; // zero is not prime - it'is not greater than 1
+        isPrime[ 1 ] = false; // one is not prime - it'is not greater than 1
+        isPrime[ 2 ] = true; // only one even prime
+        primes.add( 2 );
+        for ( int i = 4; i <= LIMIT; i += 2 ) {
+            isPrime[ i ] = false;
+        }
+        for ( int p = 3; p <= LIMIT; p += 2 ) {
+            if ( isPrime[ p ] ) {
+                primes.add( p );
+                for ( int m = 3; m * p < LIMIT; m += 2 ) {
+                    isPrime[ m * p ] = false;
+                }
+            }
+        }
+    }
+
+
+Maybe you would like to have primes in array instead of List — you can use toArray(int []) method or you find out how many prime numbers (lower than limit) there are. 
+
+
+## Conclusion
+The running time of block sieving is the same as for regular sieve of Eratosthenes (unless the size of the blocks is very small), but the needed memory will shorten to O(n‾√+S) and we have better caching results. On the other hand, there will be a division for each pair of a block and prime number from [1;n‾√], and that will be far worse for smaller block sizes. Hence, it is necessary to keep balance when selecting the constant S. We achieved the best results for block sizes between 104 and 105.
```

Comment:
Run your blog through Grammarly once so that the grammatical spelling mistakes are avoided and add your code in code-blocks otherwise it won't look good when the blog will get published